Sally7.Benchmarks/Serialization/SerializeArray.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using BenchmarkDotNet.Attributes;
+using Sally7.Internal;
+
+namespace Sally7.Benchmarks.Serialization;
+
+public static class SerializeArray
+{
+    public class SerializeUInt16
+    {
+        private readonly byte[] buffer = new byte[1024];
+
+        [ParamsSource(nameof(ValueParameters))]
+        public ushort[] Value { get; set; } = Array.Empty<ushort>();
+
+        public IEnumerable<ushort[]> ValueParameters =>
+            Enumerable.Range(1, 12).Select(l => Enumerable.Range(1, l).Select(x => (ushort)x).ToArray());
+
+        [Benchmark(Baseline = true)]
+        public int SerializeOneByOne()
+        {
+            ref var destination = ref MemoryMarshal.GetReference(buffer.AsSpan());
```

Comment:
Starting with .NET 5 you can also use
```suggestion
            ref byte destination = ref MemoryMarshal.GetArrayDataReference(buffer);
```

Reply:
Sounds like a good idea, at least for the benchmarks. I'm not sure about what frameworks to target, I intend to at least support the frameworks Microsoft supports, so I have to figure a nice way to support this across all targets. For the benchmarks I can perhaps live with just net5+ targeting, but I was already running into `Vector128` features that are available from net7+ only.